src/llms-txt/application/file/llms-txt-cron-scheduler.php

```diff
@@ -0,0 +1,76 @@
+<?php
+// phpcs:disable Yoast.NamingConventions.NamespaceName.TooLong
+namespace Yoast\WP\SEO\Llms_Txt\Application\File;
+
+use Yoast\WP\SEO\Helpers\Options_Helper;
+
+/**
+ * Responsible for scheduling and unscheduling the cron.
+ */
+class Llms_Txt_Cron_Scheduler {
+
+	/**
+	 * The name of the cron job.
+	 */
+	public const LLMS_TXT_POPULATION = 'wpseo_llms_txt_population';
+
+	/**
+	 * The options helper.
+	 *
+	 * @var Options_Helper
+	 */
+	private $options_helper;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Options_Helper $options_helper The options helper.
+	 */
+	public function __construct(
+		Options_Helper $options_helper
+	) {
+		$this->options_helper = $options_helper;
+	}
+
+	/**
+	 * Schedules the llms txt population cron.
+	 *
+	 * @return void
+	 */
+	public function schedule_weekly_llms_txt_population(): void {
+		if ( $this->options_helper->get( 'enable_llms_txt', true ) !== true ) {
+			return;
+		}
+
+		if ( ! \wp_next_scheduled( self::LLMS_TXT_POPULATION ) ) {
+			\wp_schedule_event( ( \time() + \WEEK_IN_SECONDS ), 'weekly', self::LLMS_TXT_POPULATION );
+		}
+	}
+
+	/**
+	 * Schedules the llms txt population cron.
+	 *
+	 * @return void
+	 */
+	public function schedule_quick_llms_txt_population(): void {
+		if ( $this->options_helper->get( 'enable_llms_txt', true ) !== true ) {
+			return;
+		}
+
+		if ( ! \wp_next_scheduled( self::LLMS_TXT_POPULATION ) ) {
```

Comment:
I think that for the quick population, it might made sense to delete any existing scheduled LLMS_TXT_POPULATION actions.

So, for the case where:
* User toggles the feature on
  * which means a scheduled action is set to happen in a week
* User deactivates and reactivates plugin
  * as it is now, the action will not be scheduled for the next 5 minutes, but will stay scheduled for the next week

I know that it's a super niche case that will even be fixed with https://github.com/Yoast/reserved-tasks/issues/566, but in principle, setting the quick scheduled action should overwrite the longer one, so let's consider this here.